src/extension.ts

```diff
@@ -19,6 +20,26 @@ export function activate(context: vscode.ExtensionContext) {
       "Configuration": {}
     });
   };
+  const importsCommand = async (editor, edit, {completion}) => {
+    setTimeout(async () => {
+      try {
+        let selection = editor.selection;
+        let completionSelection = new vscode.Selection(selection.active.translate(0, -completion.length), selection.active);
+        let codeActionCommands = await vscode.commands.executeCommand<vscode.CodeAction[]>('vscode.executeCodeActionProvider', editor.document.uri, completionSelection, vscode.CodeActionKind.QuickFix);
+        let importCommands = codeActionCommands.filter(c => c.title.toLocaleLowerCase().includes("import"));
+        if (importCommands.length) {
+          let [firstCommand] = importCommands;
+          await vscode.workspace.applyEdit(firstCommand.edit);
+          await vscode.commands.executeCommand(firstCommand.command.command, firstCommand.command.arguments);
+          await vscode.commands.executeCommand(COMPLETION_IMPORTS, {completion} );
+        }
+      } catch (error) {
+        console.log(error);
+      }
+    }, 400);
```

Comment:
I'm not entirely sure why we're waiting here, i'm guessing VSCode will only add the quickfix only after the line is actually added. I'm wondering if there's a way for us to grab the selection before we `setTimeout`, I'm afraid this can lead to a race condition

Reply:
the problem here is with the `vscode.executeCodeActionProvider` command that is not returning any suggestion if invoked with less than 200ms, but we definitely can grab the selection before the timeout, good catch, i'll change it 👍